src/activity_handlers.cpp

```diff
@@ -460,20 +463,19 @@ void activity_handlers::butcher_finish( player_activity *act, player *p )
         }
     }
 
-    if( pieces <= 0 ) {
-        p->add_msg_if_player(m_bad, _("Your clumsy butchering destroys the flesh!"));
-    } else {
-        p->add_msg_if_player(m_good, _("You harvest some flesh."));
-        const itype_id meat = corpse->get_meat_itype();
-        if( meat == "null" ) {
-            return;
-        }
+    //now handle the meat, if there is any
+    //bone_tainted is handled above (treated as a byproduct, not a meat)
+    if( pieces <= 0 && meat != "null" && meat != "bone_tainted" ) {
+        p->add_msg_if_player( m_bad, _( "Your clumsy butchering destroys the flesh!" ) );
+    } else if( meat!= "null" && meat != "bone_tainted" ) {
+        p->add_msg_if_player( m_good, _( "You harvest some flesh." ) );
 
         item chunk( meat, age );
         chunk.set_mtype( corpse );
 
-        // for now don't drop non-tainted parts overhaul of taint system to not require excessive item duplication
-        item parts( chunk.is_tainted() || chunk.has_flag( "CANNIBALISM" ) ? meat : "offal", age );
+        // for now don't drop tainted or cannibal. parts overhaul of taint system to not require excessive item duplication
+        bool make_offal = !chunk.is_tainted() && !chunk.has_flag( "CANNIBALISM" ) && meat != "veggy";
```

Comment:
`!chunk.made_of( material_id( "veggy" ) )` instead of `meat != "veggy"` would be better (the comment about hardcoded ids: there're much fewer materials than items).